rmw_implementation/rmw_implementation-extras.cmake.in

```diff
@@ -16,38 +16,63 @@
 
 find_package(rmw_implementation_cmake REQUIRED)
 set(default_rmw_implementation "@RMW_IMPLEMENTATION@")
+set(available_rmw_implementation "@RMW_IMPLEMENTATIONS@")
 
 if(NOT "${RMW_IMPLEMENTATION}" STREQUAL "")
-  set(requested_default_rmw_implementation ${RMW_IMPLEMENTATION})
-  if(NOT requested_default_rmw_implementation STREQUAL default_rmw_implementation)
-    message(FATAL_ERROR
-      "The default RMW implementation has been specified as "
-      "'${requested_default_rmw_implementation}' through CMake,"
-      "however the default RMW implementation which will be used is "
-      "'${default_rmw_implementation}', which was specified when the "
-      "'rmw_implementation' package was built.")
+  set(requested_rmw_implementation ${RMW_IMPLEMENTATION})
+  if(@RMW_IMPLEMENTATION_SUPPORTS_POCO@)
+    if(NOT requested_rmw_implementation IN_LIST available_rmw_implementation)
+      message(FATAL_ERROR
+        "The RMW implementation has been specified as "
+        "'${requested_rmw_implementation}' through CMake, "
+        "however it is not in the list of supported rmw implementations, "
+        "which was specified when the 'rmw_implementation' package was built.")
+    endif()
+  else()
+    if(NOT requested_rmw_implementation STREQUAL default_rmw_implementation)
+      message(FATAL_ERROR
+        "The RMW implementation has been specified as "
+        "'${requested_rmw_implementation}' through CMake, "
+        "however the RMW implementation should be "
```

Comment:
`"however the RMW implementation was already set as "`? I you want to further imply that what was specified via cmake should match this could say:

```
"however this needs to match the RMW implementation which was already set as "
"'${default_rmw_implementation}', "
"when the 'rmw_implementation' package was built.")
```